Hello @rdon-key thank you for your PR.

Looks like you need to run `go fmt` on your changes:

```shell
Unformatted:
  src/machine/machine_rp2040_usb.go
make: *** [GNUmakefile:185: fmt-check] Error 1
```

Can you please correct that and then rebase those changes into this PR? Thanks!

--- a/src/machine/machine_rp2040_usb.go
+++ b/src/machine/machine_rp2040_usb.go
@@ -169,21 +169,22 @@ func initEndpoint(ep, config uint32) {
 }
 
 func handleUSBSetAddress(setup usb.Setup) bool {
+	// Using 570μs timeout which is exactly the same as SAMD21.
+	const ackTimeout = 570
+
+	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
 	sendUSBPacket(0, []byte{}, 0)
 
-	// last, set the device address to that requested by host
-	// wait for transfer to complete
-	timeout := 3000
-	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
+	// Wait for transfer to complete with a timeout.
+	t := timer.timeElapsed()
 	for (rp.USBCTRL_REGS.SIE_STATUS.Get() & rp.USBCTRL_REGS_SIE_STATUS_ACK_REC) == 0 {
-		timeout--
-		if timeout == 0 {
-			return true
+		if dt := timer.timeElapsed() - t; dt >= ackTimeout {
+			return false
 		}
 	}
 
+	// Set the device address to that requested by host.
 	rp.USBCTRL_REGS.ADDR_ENDP.Set(uint32(setup.WValueL) & rp.USBCTRL_REGS_ADDR_ENDP_ADDRESS_Msk)
-
 	return true
 }
 

--- a/src/machine/machine_rp2350_usb.go
+++ b/src/machine/machine_rp2350_usb.go
@@ -174,10 +174,10 @@ func initEndpoint(ep, config uint32) {
 func handleUSBSetAddress(setup usb.Setup) bool {
 	// Using 570μs timeout which is exactly the same as SAMD21.
 
-	const ackTimeout=570
+	const ackTimeout = 570
 	rp.USB.SIE_STATUS.Set(rp.USB_SIE_STATUS_ACK_REC)
 	sendUSBPacket(0, []byte{}, 0)
-	
+
 	// Wait for transfer to complete with a timeout.
 	t := timer.timeElapsed()
 	for (rp.USB.SIE_STATUS.Get() & rp.USB_SIE_STATUS_ACK_REC) == 0 {
@@ -185,12 +185,11 @@ func handleUSBSetAddress(setup usb.Setup) bool {
 			return false
 		}
 	}
-	
+
 	// Set the device address to that requested by host.
 	rp.USB.ADDR_ENDP.Set(uint32(setup.WValueL) & rp.USB_ADDR_ENDP_ADDRESS_Msk)
 	return true
 }
-
 
 // SendUSBInPacket sends a packet for USB (interrupt in / bulk in).
 func SendUSBInPacket(ep uint32, data []byte) bool {

--- a/src/machine/machine_rp2350_usb.go
+++ b/src/machine/machine_rp2350_usb.go
@@ -172,23 +172,25 @@ func initEndpoint(ep, config uint32) {
 }
 
 func handleUSBSetAddress(setup usb.Setup) bool {
-	sendUSBPacket(0, []byte{}, 0)
+	// Using 570μs timeout which is exactly the same as SAMD21.
 
-	// last, set the device address to that requested by host
-	// wait for transfer to complete
-	timeout := 3000
+	const ackTimeout=570
 	rp.USB.SIE_STATUS.Set(rp.USB_SIE_STATUS_ACK_REC)
+	sendUSBPacket(0, []byte{}, 0)
+	
+	// Wait for transfer to complete with a timeout.
+	t := timer.timeElapsed()
 	for (rp.USB.SIE_STATUS.Get() & rp.USB_SIE_STATUS_ACK_REC) == 0 {
-		timeout--
-		if timeout == 0 {
-			return true
+		if dt := timer.timeElapsed() - t; dt >= ackTimeout {
+			return false
 		}
 	}
-
+	
+	// Set the device address to that requested by host.
 	rp.USB.ADDR_ENDP.Set(uint32(setup.WValueL) & rp.USB_ADDR_ENDP_ADDRESS_Msk)
-
 	return true
 }
+
 
 // SendUSBInPacket sends a packet for USB (interrupt in / bulk in).
 func SendUSBInPacket(ep uint32, data []byte) bool {

